packages/devtools_app/lib/src/inspector/inspector_service.dart

```diff
@@ -179,11 +201,115 @@ class InspectorService extends DisposableController
     return false;
   }
 
+  ValueListenable<List<String>> get rootDirectories => _rootDirectories;
+  final ValueNotifier<List<String>> _rootDirectories = ValueNotifier([]);
+
+  @visibleForTesting
+  Set<String> get rootPackages => _rootPackages;
+  Set<String> _rootPackages;
+
+  @visibleForTesting
+  List<String> get rootPackagePrefixes => _rootPackagePrefixes;
+  List<String> _rootPackagePrefixes;
+
+  Future<void> _onRootDirectoriesChanged(List<String> directories) async {
+    _rootDirectories.value = directories;
+    _rootPackages = {};
+    _rootPackagePrefixes = [];
+    for (var directory in directories) {
+      // TODO(jacobr): add an API to DDS to provide the actual mapping to and
+      // from absolute file paths to packages instead of having to guess it
+      // here.
+      assert(!directory.startsWith('package:'));
+
+      final parts =
+          directory.split('/').where((element) => element.isNotEmpty).toList();
+      final libIndex = parts.lastIndexOf('lib');
+      final path = libIndex > 0 ? parts.sublist(0, libIndex) : parts;
+      // Special case handling of bazel packages.
+      final google3Index = path.lastIndexOf('google3');
+      if (google3Index != -1 && google3Index + 1 < path.length) {
+        var packageParts = path.sublist(google3Index + 1);
+        //
+        if (packageParts[0] == 'third_party' && packageParts.length >= 3) {
```

Comment:
add a const for 3 so it is clear what that number is for

Reply:
done. Added a comment as well.